lib/AST/LayoutConstraint.cpp

```diff
@@ -106,6 +110,24 @@ bool LayoutConstraintInfo::isNativeRefCountedObject(LayoutConstraintKind Kind) {
   return Kind == LayoutConstraintKind::NativeRefCountedObject;
 }
 
+bool LayoutConstraintInfo::isClass(LayoutConstraintKind Kind) {
+  return Kind == LayoutConstraintKind::Class;
+}
```

Comment:
For this I introduced `isAnyClass`.  This predicate is intended to be used to check the exact kind.